### PR TITLE
Use __func__ for log messages

### PR DESCRIPTION
**Pull request by Bushstar**
Remove literal function names and print them using __func__. There are a few updated log messages in this commit that have been moved to a new function but still refer to their previous function, printing the function name with __func__ will resolve this in future. Using __func__ also removes logging code from grep results when searching on function name.

A previous PR to change a single mislabelled log entry was added here.

https://github.com/bitcoin/bitcoin/pull/15979

This PR includes the change from the previous one and adds every additional entry found by myself and practicalswift.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

**Review by jnewbery — COMMENTED** [review]
Concept ACK for places where this doesn't change the log message.

> There are a few updated log messages in this commit that have been moved to a new function but still refer to their previous function

There's an argument for not changing these, since any automated log parsing tools may be matching on the old log strings.

**Review by promag — COMMENTED** [review]
In some cases it's logged `class::func`, does it make sense to normalize - either always have the class or just the function?

File: src/validation.cpp [hunk 1]
```diff
@@ -4148,12 +4148,12 @@ bool CChainState::ReplayBlocks(const CChainParams& params, CCoinsView* view)
         if (pindexOld->nHeight > 0) { // Never disconnect the genesis block.
             CBlock block;
             if (!ReadBlockFromDisk(block, pindexOld, params.GetConsensus())) {
-                return error("RollbackBlock(): ReadBlockFromDisk() failed at %d, hash=%s", pindexOld->nHeight, pindexOld->GetBlockHash().ToString());
+                return error("%s: ReadBlockFromDisk() failed at %d, hash=%s", __func__, pindexOld->nHeight, pindexOld->GetBlockHash().ToString());
```
[the remark this answers is not in the document]
Comment (Bushstar) [reply src/validation.cpp]: That error was first added by sipa in the commit below and has always been RollbackBlock() even though in the ReplayBlocks function. There's a RollforwardBlock function added above ReplayBlocks so I suspect it was a WIP name that did not get updated in the error call.

https://github.com/bitcoin/bitcoin/commit/013a56aa1af985894b3eaf7c325647b0b74e4456

**Comment by Bushstar** [comment]
I personally prefer `class::func` where applicable, I'll update this PR. Normalising all logging like this is a much bigger task and will conflict with a lot of other PRs, I'll stick to the ones in this PR for now.

**Comment by Bushstar** [comment]
Travis hit the time limit, seems less prone to this in the morning UK, I'll force push again tomorrow morning if it's not been manually run and succeeded by then.

**Comment by gmaxwell** [comment]
jnewbery: that same argument pretty much argues against doing this, since all these messages will change in refactors.  I'm not sure how we ended up with so many function names inside log entries, it isn't generally a good idea: we shouldn't have our errors change due to refactoring. Not just log parsing tools, but also users googling messages.

**Comment by jnewbery** [comment]
> that same argument pretty much argues against doing this, since all these messages will change in refactors.

Right, so what do you suggest? Removing the function names from logs entirely?

**Comment by MarcoFalke** [comment]
Most of the logs are error conditions and there are two parts of the log message with different purpose:
* The function name/class name to tell the developer where it happened
* The verbose text to explain what happened

I think search engines today can deal with a single word that is off. And including more information for developers can't hurt, as long as it is correct. Not sure how many automated log parsers parse for the error strings including the location...

**Comment by promag** [comment]
I think we should feel free to change logs, it isn't like an API, is it?

**Comment by gmaxwell** [comment]
Changing of the function name will reliably break log parsing. Rather than including potentially changing function names the messages should be descriptive enough that there isn't any trouble finding them (considering how often functions move from file to file it isn't like a search is avoidable!). Then both needs are satisfied.

**Comment by MarcoFalke** [comment]
Couldn't the logs use something like `__FILE__` and `__LINE__`. In combination with the commit from the debug log, this should be sufficient to find the exact location without searching.

**Comment by sipa** [comment]
@gmaxwell For new code, with new messages, I'm very much in favor of using descriptive searchable and not-code-organization-dependent message texts. But what do you suggest we do with the code that is already there right now? Messages with function names in them have been in the codebase since before the project was on Github. We can replace them, and break the searchability property once, or we can hardcode them and remain searchable but confusing...

**Comment by gmaxwell** [comment]
@sipa I could go either way-- but using __func__ is not either of those ways.  I would personally prefer that we replace them with strings that don't need to be changed unless the underlying behaviour changes.  I would rather that they be potentially misleading then generated by __func__.  No one working on the code itself is going to be confused too long by a log entry with an incorrect function name in it: the first thing you do in the code when dealing with an error is find the line that generated it, and if it's not there then its somewhere else.

As an aside, __file__ __line__ macros turn a much wider class of changes into binary changes, already when reviewing stuff for being cosmetic only I change macros, having them peppered all over the codebase would make that much harder.   Also, it will further cause random failures in log file comparisons: joe user or operations guy doesn't realize that the number in the message is unstable and will differ from version to version.

**Comment by laanwj** [comment]
I think this is way too controversial for a simple logging change, so I'm going to close this, sorry.

>  I'm not sure how we ended up with so many function names inside log entries, it isn't generally a good idea: we shouldn't have our errors change due to refactoring

I tend to agree with this. It'd make much more sense to prefix messages with a broad category (net, rpc, validation, ...) than the function name which leaks a lot of implementation details for no good reason.

**Pull request closed without merging**